File: Java Solutions/reverse_integer.java

public class reverse_integer {
	public static int reverse(int x) {
		// if x is negative
		if (x < 0) {
			String num = String.valueOf(x);
			String reversedNumber = "";
			int length = num.length();
			int len = num.length() - 1;
			reversedNumber += "-";
			int counter = 1;
			while (counter < length) {
				reversedNumber += num.charAt(len);
				len--;
				counter++;
			}

			try {
				// System.out.println(reversedNumber);
				int s = Integer.parseInt(reversedNumber);

				if (s < (-2e31)) {
					return 0;
				} else {
					return s;
				}

			} catch (Exception e) {

			}

			// if x is positive
		} else {

			String num = String.valueOf(x);
			String reversedNumber = "";
			int length = num.length();
			int len = num.length() - 1;
			int counter = 0;
			while (counter < length) {
				reversedNumber += num.charAt(len);
				len--;
				counter++;
			}

			try {
				// System.out.println(reversedNumber);
				int s = Integer.parseInt(reversedNumber);

				if (s > ((2e31) - 1)) {
					return 0;
				} else {
					return s;
				}

			} catch (Exception e) {

			}

		}
		return 0;

	}
}

File: reverse_integer.py
class Solution(object):
    def reverse(self, x):

        x_string = str(x)

        if len(x_string) > 1:
            x_string = x_string[::-1]
            counter = 0

            while counter < len(x_string):
                if x_string[counter] != "0":
                    break
                counter+=1

            x_string = x_string[counter:]

            if len(x_string) > 0:
                if x_string[-1] == "-":
                    x_string = x_string[-1] + x_string[0:-1]

        if len(x_string):
            x = int(x_string)
            if x >= -2**31 and x <= (2**31)-1:
                return x
            else:
                return 0

